File: TemplateMethodPattern/src/com/company/kainosSkaiciavimas/Ikainiai.java
package com.company.kainosSkaiciavimas;

import java.math.BigDecimal;

public class Ikainiai {
    private final BigDecimal pradineKaina;
    private final BigDecimal kilometroKaina;
    private final BigDecimal minutesKaina;
    private final BigDecimal virsytoLaikoKaina;
    private final BigDecimal virsytoAtstumoKaina;
    private final BigDecimal parosKaina;
    private final BigDecimal savaitesKaina;
    private final BigDecimal menesioKaina;

    /**
     * Kainos turi buti nurodytos centais.
     */
    public Ikainiai(int pradineKaina, int kilometroKaina, int minutesKaina, int virsytoLaikoKaina, int virsytoAtstumoKaina, int parosKaina, int savaitesKaina, int menesioKaina) {
        this.pradineKaina = new BigDecimal(pradineKaina);
        this.kilometroKaina = new BigDecimal(kilometroKaina);
        this.minutesKaina = new BigDecimal(minutesKaina);
        this.virsytoLaikoKaina = new BigDecimal(virsytoLaikoKaina);
        this.virsytoAtstumoKaina = new BigDecimal(virsytoAtstumoKaina);
        this.parosKaina = new BigDecimal(parosKaina);
        this.savaitesKaina = new BigDecimal(savaitesKaina);
        this.menesioKaina = new BigDecimal(menesioKaina);
    }

    public BigDecimal getPradineKaina() {
        return pradineKaina;
    }

    public BigDecimal getKilometroKaina() {
        return kilometroKaina;
    }

    public BigDecimal getMinutesKaina() {
        return minutesKaina;
    }

    public BigDecimal getVirsytoLaikoKaina() {
        return virsytoLaikoKaina;
    }

    public BigDecimal getVirsytoAtstumoKaina() {
        return virsytoAtstumoKaina;
    }

    public BigDecimal getParosKaina() {
        return parosKaina;
    }

    public BigDecimal getSavaitesKaina() {
        return savaitesKaina;
    }

    public BigDecimal getMenesioKaina() {
        return menesioKaina;
    }
}

File: Mixins/src/com/company/kainosSkaiciavimas/Ikainiai.scala
package com.company.kainosSkaiciavimas

import java.math.BigDecimal

/**
 * Kainos turi buti nurodytos centais.
 */
class Ikainiai(_pradineKaina: Int, _kilometroKaina: Int, _minutesKaina: Int, _virsytoLaikoKaina: Int,
               _virsytoAtstumoKaina: Int = 0, _parosKaina: Int = 0, _savaitesKaina: Int = 0, _menesioKaina: Int = 0) {

  def pradineKaina : BigDecimal = new BigDecimal(_pradineKaina)

  def kilometroKaina : BigDecimal = new BigDecimal(_kilometroKaina)

  def minutesKaina : BigDecimal = new BigDecimal(_minutesKaina)

  def virsytoLaikoKaina : BigDecimal = new BigDecimal(_virsytoLaikoKaina)

  def virsytoAtstumoKaina : BigDecimal = new BigDecimal(_virsytoAtstumoKaina)

  def parosKaina : BigDecimal = new BigDecimal(_parosKaina)

  def savaitesKaina : BigDecimal = new BigDecimal(_savaitesKaina)

  def menesioKaina : BigDecimal = new BigDecimal(_menesioKaina)
}